### PR TITLE
Rename CrateAction>Notifications to Sound and add an actual Notifications field.

### LINKED ISSUE #15698: Add support for actual CrateAction Notifications
Currently there is a `Notifications:` value under CrateAction but it actually plays a sound. Sole Survivor needs actual notifications for each crate bonus.

### PR DESCRIPTION
**Pull request by MustaphaTR**
Closes #15698.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Crates/CrateAction.cs [hunk 1]
```diff
@@ -73,7 +76,11 @@ public virtual int GetSelectionShares(Actor collector)
 
 		public virtual void Activate(Actor collector)
 		{
-			Game.Sound.PlayToPlayer(SoundType.World, collector.Owner, info.Notification);
+			Game.Sound.PlayToPlayer(SoundType.World, collector.Owner, info.Sound);
```
[the remark this answers is not in the document]
Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/Crates/CrateAction.cs]: This has been like that (and TD and D2k don't use sounds at all) so i'm gonna guess there is already a check in `PlayToPlayer`.

File: mods/ts/rules/misc.yaml [hunk 2]
```diff
@@ -89,21 +89,21 @@ CRATE:
...
 	GrantExternalConditionCrateAction@armor:
 		SelectionShares: 5
 		Effect: armor
 		Condition: crate-damage
-		Notification: 00-i068.aud
+		Sound: 00-i068.aud
 	GrantExternalConditionCrateAction@speed:
 		SelectionShares: 5
 		Condition: crate-speed
-		Notification: 00-i080.aud
```
Comment (abcdefg30) [inline mods/ts/rules/misc.yaml]: Aren't those real speech notifications?

File: OpenRA.Mods.Common/Traits/Crates/CrateAction.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Crates/CrateAction.cs]: There is a `string,IsNullOrEmpty` check, yes.

[the remark this answers is not in the document]
Comment (MustaphaTR) [reply OpenRA.Mods.Common/Traits/Crates/CrateAction.cs]: Thinkg about this one, should i change this to play to everyone? This is now actually meant for sounds, not notifications which makes more sense to play to everyone.

[the remark this answers is not in the document]
Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/Crates/CrateAction.cs]: Yes, that makes sense. :+1:

File: OpenRA.Mods.Common/UpdateRules/Rules/20180923/RenameCrateActionNotification.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+{
+	public class RenameCrateActionNotification : UpdateRule
+	{
+		public override string Name { get { return "*CrateAction>Notification renamed to Sound"; } }
+		public override string Description
+		{
+			get
+			{
+				return "*CrateAction: traits now have actual `Notification:` field.\n" +
+					"New `Sound:` field does what old `Notification:` did.";
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/UpdateRules/Rules/20180923/RenameCrateActionNotification.cs]: I'd suggest adding some articles and omitting the colons:
```
return "'*CrateAction' traits now have an actual `Notification` field.\n" +
	"The new `Sound` field does what the old `Notification` did.";
```

**Review by abcdefg30 — DISMISSED** [review]
Looks good to me otherwise. :+1:

**Comment by MustaphaTR** [comment]
Updated.

**Comment by abcdefg30** [comment]
Needs a rebase again...

**Comment by MustaphaTR** [comment]
Rebased.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/4d330ccce13bc9bd5aea7499f96390a306e8c8e2

**Comment by MustaphaTR** [comment]
It seems like i forgot to put the update rule in the UpdatePath.cs.

**Comment by pchote** [comment]
Can you please file a PR to fix that?

**Pull request merged**